src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,346 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::CUBE:
+  case visualization_msgs::Marker::CYLINDER:
+  case visualization_msgs::Marker::SPHERE:
+    checkQuaternion(marker, ss);
+    checkScale(marker, ss);
+    checkColor(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+
+  case visualization_msgs::Marker::LINE_STRIP:
+  case visualization_msgs::Marker::LINE_LIST:
+    checkQuaternion(marker, ss);
+    checkScaleLineStripAndList(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::SPHERE_LIST:
+  case visualization_msgs::Marker::CUBE_LIST:
+  case visualization_msgs::Marker::TRIANGLE_LIST:
+    checkQuaternion(marker, ss);
+    checkScale(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::POINTS:
+    checkScalePoints(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::TEXT_VIEW_FACING:
+    checkColor(marker, ss);
+    checkScaleText(marker, ss);
+    checkTextNotEmptyOrWhitespace(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::MESH_RESOURCE:
+    checkQuaternion(marker, ss);
+    checkColor(marker, ss);
+    checkScale(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    break;
+
+  default:
+    ss << "Unknown marker type: " << marker.type ;
+  }
+
+  if(ss.tellp() != 0) //stringstream is not empty
+  {
+    std::string warning = ss.str();
+
+    owner->setMarkerStatus(MarkerID(marker.ns, marker.id), StatusProperty::Warn, warning);
+    ROS_WARN("Marker '%s/%d': %s", marker.ns.c_str(), marker.id, warning.c_str());
+    return false;
+  }
+
+  owner->deleteMarkerStatus(MarkerID(marker.ns, marker.id));
+  return true;
+}
+
+bool checkMarkerArrayMsg(const visualization_msgs::MarkerArray& array, MarkerDisplay* owner)
+{
+  std::vector<MarkerID> marker_ids;
+
+  bool add_marker_in_array = false;
+
+  for(int i = 0; i<array.markers.size(); i++)
+  {
+    if(array.markers[i].action == visualization_msgs::Marker::ADD)
+    {
+      add_marker_in_array = true;
+    }
+
+    if(add_marker_in_array && i != 0 && array.markers[i].action == visualization_msgs::Marker::DELETEALL)
+    {
+      std::stringstream warning;
+      warning << "found a DELETEALL at index " << i << ", previous markers in the MarkerArray will never show";
+      ROS_WARN("MarkerArray: %s", warning.str().c_str());
```

Comment:
Calling warning.str() twice in a row here, will allocate memory for the string twice.

Reply:
Single allocation is done in b8ef09d